### PR TITLE
Adds engine option to `LazyReferenceMapper`

### PR DESCRIPTION
**Pull request by norlandrhagen**
Defaults to 'fastparquet', but allows for 'pyarrow'

### CODE REVIEW

**Comment by martindurant** [comment]
Are you aware of arro3, a new minimalist arrow implementation that does include a parquet reader? Maybe would be perfect for this, but perhaps still too immature to use.

**Comment by norlandrhagen** [comment]
> Are you aware of arro3, a new minimalist arrow implementation that does include a parquet reader? Maybe would be perfect for this, but perhaps still too immature to use.

I just stumbled on [that](https://github.com/kylebarron/arro3) yesterday! Seems like a great idea, but yeah, it might be a bit too early to use 🤷

File: fsspec/implementations/reference.py [hunk 1]
```diff
@@ -126,11 +132,20 @@ def __init__(
             Encode urls as pandas.Categorical to reduce memory footprint if the ratio
             of the number of unique urls to total number of refs for each variable
             is greater than or equal to this number. (default 10)
+        engine: Literal["fastparquet","pyarrow"]
+            Engine choice for reading parquet files. (default is "fastparquet")
         """
+
+        from importlib.util import find_spec
+
+        if find_spec("pyarrow") is None:
```
Comment (martindurant) [inline fsspec/implementations/reference.py:141]: This gets called even if engine is "fastparquet", no? That would make this unusable if pyarrow isn't installed.

File: fsspec/implementations/reference.py [hunk 2]
```diff
@@ -463,18 +478,28 @@ def write(self, field, record, base_url=None, storage_options=None):
 
         fn = f"{base_url or self.out_root}/{field}/refs.{record}.parq"
         self.fs.mkdirs(f"{base_url or self.out_root}/{field}", exist_ok=True)
+
+        if self.engine == "pyarrow":
+            df_backend_kwargs = {}
```
Comment (martindurant) [inline fsspec/implementations/reference.py:483]: There are probably some good defaults here?

File: fsspec/implementations/reference.py (hunk 1 above)
Comment (martindurant) [reply fsspec/implementations/reference.py:141]: (could test by monkeypatch on find_spec)

Comment (norlandrhagen) [reply fsspec/implementations/reference.py:141]: Great catch!

File: fsspec/implementations/reference.py (hunk 2 above)
Comment (norlandrhagen) [reply fsspec/implementations/reference.py:483]: Good call! I added a few, I'll see how the tests go :) 

Absolutely not a parquet expert, so we'll see.

**Pull request merged**